### PR TITLE
[expo-dev-menu][expo-dev-launcher] Disable onboarding popup with URL query param

### PR DESCRIPTION
**Pull request by douglowder**
# Why

The onboarding popup breaks some UI tests (e.g. Detox), so there should be an easy way to disable it without having to rebuild the app.

# How

Add code to check for `disableOnboarding` in the URL and set `DevMenuPreferences.isOnboardingFinished=true` before the app launches.

Additionally, if testing in Detox, you can pass in a launch param to disable the onboarding popup in iOS (but not in Android) as in this example:

```js
        await device.launchApp({
          newInstance: true,
          launchArgs: {
            EXDevMenuIsOnboardingFinished: true,
          },
        });
```

# Test Plan

Tested by passing in a URL like `http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&disableOnboarding=1` in the dev launcher.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuDevSettings.kt [hunk 1]
```diff
@@ -23,15 +23,18 @@ object DevMenuDevSettings {
...
         putBoolean("isHotLoadingAvailable", devSettings.isJSDevModeEnabled)
         putBoolean("isPerfMonitorAvailable", devSettings.isJSDevModeEnabled)
-        putBoolean("isJSInspectorAvailable", run {
-          val packageName = reactInstanceManager.currentReactContext?.packageName
-            ?: return@run false
-          val metroHost = "http://${devSettings.packagerConnectionSettings.debugServerHost}"
-          runBlocking {
-            DevMenuManager.metroClient
-              .queryJSInspectorAvailability(metroHost, packageName)
+        putBoolean(
```
Comment (douglowder) [inline packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuDevSettings.kt:26]: This change is from running `spotlessApply`

File: packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/DevMenuActivity.kt [hunk 2]
```diff
@@ -1,7 +1,5 @@
 package expo.modules.devmenu
 
-import android.content.Context
```
Comment (douglowder) [inline packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/DevMenuActivity.kt:3]: This change is from running `spotlessApply`

File: packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/extensions/DevMenuExtension.kt [hunk 3]
```diff
@@ -15,7 +15,6 @@ import expo.interfaces.devmenu.items.KeyCommand
 import expo.modules.devmenu.DEV_MENU_TAG
 import expo.modules.devmenu.DevMenuManager
 import expo.modules.devmenu.devtools.DevMenuDevToolsDelegate
-import kotlinx.coroutines.runBlocking
```
Comment (douglowder) [inline packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/extensions/DevMenuExtension.kt:18]: This change is from running `spotlessApply`

File: packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/modules/DevMenuInternalModule.kt [hunk 4]
```diff
@@ -1,6 +1,5 @@
 package expo.modules.devmenu.modules
 
-import android.os.Build
```
Comment (douglowder) [inline packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/modules/DevMenuInternalModule.kt:3]: This change is from running `spotlessApply`

File: packages/expo-dev-menu/android/src/release/java/expo/modules/devmenu/DevMenuManager.kt [hunk 5]
```diff
@@ -96,7 +96,6 @@ object DevMenuManager : DevMenuManagerInterface {
     throw IllegalStateException(DEV_MENU_IS_NOT_AVAILABLE)
   }
 
-
```
Comment (douglowder) [inline packages/expo-dev-menu/android/src/release/java/expo/modules/devmenu/DevMenuManager.kt:99]: This change is from running `spotlessApply`

File: packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift [hunk 6]
```diff
@@ -52,6 +53,18 @@ public class EXDevLauncherURLHelper: NSObject {
...
+  @objc
+  public static func disableOnboardingPopupIfNeeded(_ url: URL) {
+    let components = URLComponents.init(url: url, resolvingAgainstBaseURL: false)
+
+    for queryItem in components?.queryItems ?? [] {
+      if queryItem.name == "disableOnboarding" && queryItem.value != nil {
+        DevMenuPreferences.isOnboardingFinished = true
+        break
+      }
+    }
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift:65]:
```suggestion
    if components?.queryItems?.contains(where: { $0.name == "disableOnboarding" && $0.value != nil }) {
        DevMenuPreferences.isOnboardingFinished = true
    }
```

Comment (lukmccall) [reply packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift:65]: Also, I think we should check the exact value of the `disableOnboarding` parameter and compere it to 1.

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift (hunk 6 above)
Comment (douglowder) [reply packages/expo-dev-launcher/ios/EXDevLauncherURLHelper.swift]: Made the change to this syntax (also did this for the other method above, and made it check for the value 1. Ran `swiftlint --fix` on this file as well.

**Review by esamelson — APPROVED** [review]

File: packages/expo-dev-menu/ios/Modules/DevMenuPreferences.swift [hunk 7]
```diff
@@ -84,8 +84,10 @@ public class DevMenuPreferences: NSObject, RCTBridgeModule {
 
   /**
    Returns `true` only if the user finished onboarding, `false` otherwise.
+   This is now public because expo-dev-launcher needs access
```
Comment (ajsmth) [inline packages/expo-dev-menu/ios/Modules/DevMenuPreferences.swift:87]:
```suggestion
   This is public because expo-dev-launcher needs access
```

**Review by ajsmth — APPROVED** [review]
Looks good! Personally I think it makes more sense to have a separate flag for disabled rather than turning off `isOnboardingFinished` in this case, but I don't think it makes much of a difference. 

Is there some instructions / plans for documentation around this? I'm not 100% familiar with detox configurations, it might be nice to add some info around this

File: packages/expo-dev-menu/ios/Modules/DevMenuPreferences.swift (hunk 7 above)
Comment (douglowder) [reply packages/expo-dev-menu/ios/Modules/DevMenuPreferences.swift:87]: @ajsmth I am actively working on the docs for this -- had a discussion with @brentvatne yesterday and we are attempting to figure out exactly where this should be in the docs ("Development Builds" and/or "EAS E2E tests").

I'll go ahead and land this and create a separate PR for the doc changes.

**Comment by pradeipp** [comment]
Hi @douglowder I saw your description and was wondering - will this change not work on android?

**Comment by douglowder** [comment]
@pradeipp this change works correctly on Android. The issue with Android is just with Detox testing, and is still under investigation.

**Comment by pradeipp** [comment]
great, thank you @douglowder is there an issue for it that I can follow?

**Comment by douglowder** [comment]
https://github.com/expo/eas-tests-example/issues/4

**Pull request merged**